tests/utils/test_eloop.c

```diff
@@ -0,0 +1,169 @@
+#define _GNU_SOURCE
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <string.h>
+#include <inttypes.h>
+#include <unistd.h>
+#include <setjmp.h>
+#include <stdint.h>
+#include <cmocka.h>
+
+#include "utils/log.h"
+#include "utils/eloop.h"
+#include "utils/sockctl.h"
+
+#define TMP_PFX "tdoms"
+#define TEST_ELOOP_PARAM "param"
+#define TEST_SEND_BUF_DATA "test"
+
+void test_eloop_sock_handler_read(int sock, void *eloop_ctx, void *sock_ctx) {
+  struct eloop_data *eloop = (struct eloop_data *)eloop_ctx;
+  char *sock_ctx_data = (char *)sock_ctx;
+
+  assert_non_null(eloop);
+  assert_string_equal(sock_ctx_data, TEST_ELOOP_PARAM);
```

Comment:
It's possible to use the CMocka's [Check Parameters](https://api.cmocka.org/group__cmocka__param.html) API to test for these in a generic way, but not really necessary.

Reply:
Good point.